presto-password-authenticators/src/main/java/io/prestosql/plugin/password/file/EncryptionUtil.java

```diff
@@ -0,0 +1,117 @@
+import com.google.common.base.Splitter;
+import com.google.common.io.BaseEncoding;
+import io.airlift.log.Logger;
+import io.prestosql.spi.PrestoException;
+
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.List;
+
+import static io.prestosql.spi.StandardErrorCode.CONFIGURATION_INVALID;
+
+public class EncryptionUtil
+{
+    private static final Logger log = Logger.get(EncryptionUtil.class);
+
+    private EncryptionUtil() {}
+
+    // enforce minumum cost in for BCrypt Passwords : Min value default: 8
+    public static boolean checkFormatBCryptPassword(String password, int limit)
+    {
+        try {
+            BCrypt.HashData hashData = BCrypt.Version.VERSION_2A.parser.parse(password.getBytes());
+            return hashData.cost >= limit;
+        }
+        catch (IllegalBCryptFormatException e) {
+            log.error("Error while checking for bcrypt password format " + e);
+        }
+        throw new PrestoException(CONFIGURATION_INVALID, "Invalid bcrypt Cost: %s (The minimum allowed BCrypt Cost should be 8)");
+    }
+
+    // enforce minimum iterations for PBJKDF2 passwords : Min Value defult: 1000
+    public static boolean checkFormatPBKDF2Password(String password, int limit)
+    {
+        List<String> pass = Splitter.on(":").limit(2).splitToList(password);
+        return Integer.parseInt(pass.get(0)) >= limit;
+    }
+
+    public static boolean checkBCryptPassword(String inputPassword, String hashedPassword)
+    {
+        return BCrypt.verifyer().verify(inputPassword.toCharArray(), hashedPassword).verified;
+    }
+
+    public static boolean checkPBKDF2Password(String inputPassword, String hashedPassword)
+    {
+        if (!hashedPassword.contains(":")) {
+            return false;
+        }
+        String[] parts = hashedPassword.split(":");
+        int iterations = Integer.parseInt(parts[0]);
+
+        try {
+            byte[] salt = fromHex(parts[1]);
+            byte[] hash = fromHex(parts[2]);
+
+            PBEKeySpec spec = new PBEKeySpec(inputPassword.toCharArray(), salt, iterations, hash.length * 8);
+            SecretKeyFactory key = SecretKeyFactory.getInstance("PBKDF2WithHmacSHA1");
+            byte[] inputHash = key.generateSecret(spec).getEncoded();
+            int diff = 0;
+            diff = hash.length ^ inputHash.length;
+            if (diff != 0) { //immediate error if the hash length doesn't match
+                throw new PrestoException(CONFIGURATION_INVALID, "PBKDF2 password input is malformed");
+            }
+            for (int i = 0; i < hash.length && i < inputHash.length; i++) {
+                diff |= hash[i] ^ inputHash[i];
+            }
+            return diff == 0;
+        }
+        catch (NoSuchAlgorithmException | InvalidKeySpecException e) {
+            log.error("PBKDF2 Validation Error: " + e);
```

Comment:
Remove the logging. See above about log-and-throw

Reply:
Done